pages/api/teammatching/subscription/delete.ts

```diff
@@ -0,0 +1,68 @@
+import { firestore } from 'firebase-admin';
+import { NextApiRequest, NextApiResponse } from 'next';
+import initializeApi from '../../../../lib/admin/init';
+import { userIsAuthorized } from '../../../../lib/authorization/check-authorization';
+
+initializeApi();
+const db = firestore();
+
+// interface of subscription data that contains the user id and the post id
+interface SubscriptionData {
+  userId: string;
+  postId: string;
+}
+
+// delete one subscription for user that unsubscribe them to a post
+async function deleteSubscription(req: NextApiRequest, res: NextApiResponse) {
+  try {
+    const subscriptionData: SubscriptionData = JSON.parse(req.body);
+    const snapshot = await db
+      .collection('subscriptions')
+      .where('userId', '==', subscriptionData.userId)
+      .where('postId', '==', subscriptionData.postId)
+      .get();
+    if (snapshot.empty) {
+      return res.status(404).json({
+        msg: 'Subscription to posting not found',
+      });
+    }
+    snapshot.forEach((doc) => {
+      doc.ref.delete();
```

Comment:
I think delete returns a Promsie. Should we add await here? But since we are sure that this is a one-to-one relationship, we can take the first found doc and await for delete on that doc.

Reply:
@uy-seng

Can you change it one more time? 
More information in this: https://masteringjs.io/tutorials/fundamentals/async-foreach
Generally, using async inside forEach is not the right way to handle promises

About my other questions, do you have any answers, or should we discuss them with Mike too?

Reply:
i updated it using Promise.all instead, and sorry i just saw the other question, i think we can you the types that you created instead but need Mike input on that.